core_types/src/error/error.rs

```diff
@@ -157,6 +164,33 @@ impl From<FutureCanceled> for HolochainError {
     }
 }
 
+#[derive(Serialize, Deserialize, Default, Debug, DefaultJson)]
+pub struct ZomeApiInternalResult {
+    pub ok: bool,
+    pub value: String,
+    pub error: String,
+}
```

Comment:
Since this is the structure of the JSON that is passed through memory, it is something other HDKs need to implement as well. So far, we had all of those structs in wasm-utils/api_serialization.

I'm not sure if I want to request to move this over there for the sake of HDK developers finding it easily.

I don't want to spawn long discussions in this already big PR that has lots of awesome that I want to have merged in asap. So just marking it here for future discussion/changes, @thedavidmeister, @ddd-mtl, @Connoropolous, @willemolding.

Reply:
@ddd-mtl, oh I like this ZomeApiInternalResult and how it's used a lot. I was just talking about where it is. I think having this as the standard way makes understanding the zome API much easier. And especially implementing other HDKs. But therefore it should be somewhere where it's understood as this standard right away..

Why would you need to revert this?